api-lib/sendSocialMessage.ts

```diff
@@ -131,23 +135,40 @@ export async function sendSocialMessage({
 
   const { circles_by_pk: circle } = await queries.getCircle(circleId);
 
-  if (channels?.isDiscordBot && channels.discordBot) {
-    const { type } = channels.discordBot || {};
-    const res = await fetch(
-      `https://coordinape-discord-bot.herokuapp.com/api/epoch/${type}`,
-      {
-        method: 'POST',
-        body: JSON.stringify(channels.discordBot),
-        headers: {
-          'Content-Type': 'application/json',
-          'x-coordinape-bot-secret': COORDINAPE_BOT_SECRET,
-        },
-      }
-    );
+  const requests = [];
 
-    if (!res.ok) {
-      throw new Error(JSON.stringify(await res.json()));
-    }
+  if (channels?.isDiscordBot && channels?.discordBot) {
+    const { type } = channels.discordBot || {};
+    const updateDiscordBot = fetch(`${DISCORD_BOT_EPOCH_URL}${type}`, {
```

Comment:
there's quite a bit of duplication in the code (fetch..catch..error..json..) in these three blocks. can you make a function that takes parameters like the URL and a label like "discord-bot", to reduce the duplication?

Reply:
btw, when you make this function, please use `async() => { try { await ... } catch }` syntax instead of `.then and `.catch`